File: bridgeAddon/addon/globalPlugins/nvdaMcpBridge/adapters/ports/transport.py
# nvdaMcpBridge adapters -- the Transport seam.
# Copyright (C) 2026 Marlon Brandao de Sousa. GPL-2. See COPYING.txt.
#
# ROLE: raw byte pipe. An ADAPTER SEAM, not a domain port -- the domain speaks
#       whole messages (see domain/ports/message_channel.py) and never sees
#       bytes, so this interface lives with the adapters that use it.
# USED BY: adapters/json_lines_channel.py (frames messages over it).
# IMPLEMENTED BY: adapters/socket_transport.py (leaf, real socket; session C);
#                 tests/fakes/transport.py FakeTransport (scripted bytes).

from __future__ import annotations

from abc import ABC, abstractmethod


class Transport(ABC):
	"""A byte pipe with a poll timeout."""

	@abstractmethod
	def recv(self) -> bytes:
		"""Next chunk; ``b""`` at EOF; raises ``TimeoutError`` when idle.

		A real socket with ``settimeout`` already behaves exactly this way, which
		is why the leaf implementation is almost nothing.
		"""

	@abstractmethod
	def sendall(self, data: bytes) -> None: ...

	@abstractmethod
	def close(self) -> None: ...

File: server/adapters/ports/transport.go
// screenreader-mcp adapters -- the Transport seam.
// Copyright (C) 2026 Marlon Brandao de Sousa. GPL-2. See COPYING.txt.
//
// ROLE: adapter seam between adapters, invisible to the domain.
// IMPLEMENTED BY: adapters/bridge/net_transport.go, adapters/bridge/local_transport_windows.go, and fakes/transport.go.
// USED BY: adapters/bridge/json_lines_client.go.
package ports

import (
	"io"
	"time"
)

// Transport's Read polls: every leaf applies a short read deadline and reports an idle read as os.ErrDeadlineExceeded; io.EOF means the peer is gone.
type Transport interface {
	io.ReadWriteCloser
}

type Dialer func() (Transport, error)

// PollInterval is how long a leaf lets a Read sit idle before reporting os.ErrDeadlineExceeded.
const PollInterval = 50 * time.Millisecond
